taichi/program/program.h

```diff
@@ -94,7 +94,8 @@ class TI_DLL_EXPORT Program {
  public:
   using Kernel = taichi::lang::Kernel;
   Callable *current_callable{nullptr};
-  CompileConfig config;
+  std::unordered_map<std::thread::id, CompileConfig> configs;
```

Comment:
hmmm I guess we have an assumption here that each thread just mutates its own Compileconfig so that we didn't have any locks (good to put it in the comment) .
I wonder if thread local storage is a better choice than this?
In fact what's even better is that we can force constant folding to use its own copy of Compileconfig to compile?

Reply:
Yeah, but this will include many refactors... maybe we can do it when we implement thread-safe Taichi...

Reply:
Yep I'm fine with this as a temp workaround, would you mind adding some comments explaining that this why we have this & in the future we'd like to fix constant folding instead? Thanks!